rest/data/commit_queue.go

```diff
@@ -388,3 +488,30 @@ func GetAdditionalPatches(patchId string) ([]string, error) {
 		Message:    errors.Errorf("patch '%s' not found in commit queue", patchId).Error(),
 	}
 }
+
+func requireSigned(ctx context.Context, userRepo UserRepoInfo, prNum int) error {
+	settings, err := evergreen.GetConfig()
+	if err != nil {
+		return errors.Wrap(err, "getting admin settings")
+	}
+
+	githubToken, err := settings.GetGithubOauthToken()
+	if err != nil {
+		return errors.Wrap(err, "getting GitHub OAuth token from settings")
+	}
+
+	commits, err := thirdparty.GetGithubPullRequestCommits(ctx, githubToken, userRepo.Owner, userRepo.Repo, prNum)
+	if err != nil {
+		return errors.Wrap(err, "getting GitHub commits")
+	}
+
+	for _, c := range commits {
+		commit := c.GetCommit()
+		if commit.Verification != nil && !utility.FromBoolPtr(commit.Verification.Verified) &&
```

Comment:
Are these conditions supposed to be `||` instead of `&&`? It seems like this function should error if any of the conditions are false because any one of them might indicate the commit is not signed.

Reply:
Hmm, I'm not sure what the intention was when this function was created but that makes sense to me as well

Reply:
Hmm, not sure the context on the original function when it was written, but this makes sense to me as well. Going to change to `||`

Reply:
based on our logging, it looks like we are encountering empty commits on a pR that only has one commit and it's not empty. We may want to investigate that as well, outside of this PR.